### PR TITLE
Add TypeScript interface definitions

### PR DESCRIPTION
**Pull request by ukstv**
Fix https://github.com/ethereumjs/rlp/issues/24

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+declare module 'rlp' {
+    import { Buffer } from 'safe-buffer'
+
+    type EncodeScalarInput = Buffer|string|number
+    type EncodeInput = EncodeScalarInput|Array<EncodeScalarInput>
+
+    export function encode(input: EncodeInput): Buffer
+    export function decode(input: EncodeInput): Array<Buffer>
```
Comment (axic) [inline index.d.ts]: `decode` seems to have an optional parameter for `stream`, which means it returns a "streaming output", e.g. a the decoded data and the remainder buffer.

I'm fine not exposing it as that complicates the definition and not sure if anybody uses it.

**Comment by axic** [comment]
Need to update the definition after #16 is merged.

**Comment by holgerd77** [comment]
https://github.com/ethereumjs/rlp/pull/16 has now been merged, this needs an update.

**Comment by ukstv** [comment]
@holgerd77 Updated.

**Comment by holgerd77** [comment]
Hi @ukstv, thanks for keeping on with this. You have updated the ``standard`` library with this PR, which leads to Travis tests failing due to updated linting rules, can you do the changes to have the tests pass (or alternatively remove the library update in ``package.json``, would prefer the first solution though).

Also I am irritated why your forked ``master`` branch https://github.com/machinomy/rlp/ is displayed as being 7 commits behind while this is not shown here in the PR. Do you know why this is?

**Comment by holgerd77** [comment]
Hi @ukstv, can you reply to the latest comment? Would be great if we get this merged!

**Comment by axic** [comment]
Would it make sense merging this first?

**Comment by krzkaczor** [comment]
@axic no i don't think so. We are almost done. I will check compatibility between these typings and typings generated after our rewrite.

**Comment by ukstv** [comment]
Closing the PR, as TS rewrite is in progress.

**Comment by holgerd77** [comment]
Hi @ukstv, sorry that we couldn't merge! This PR very much had it's place though, since it was one of various triggers to start the thought process on doing a TypeScript rewrite of the whole library (with likely others to follow), so it very much had it's place, thanks again for submitting!

**Comment by ukstv** [comment]
No worries. The rewrite is awesome. Hope more of ethereumjs goes this way. BTW, is there any chance for ethereumjs to adopt TS version of truffle-hdwallet-provider? :P

**Comment by holgerd77** [comment]
@ukstv Sorry, have no deeper knowledge on this library, maybe ask @benjaminburns on that.

**Comment by holgerd77** [comment]
//cc @benjamincburns (see above)

**Pull request closed without merging**